code/core/src/manager/components/sidebar/SidebarBottom.tsx

```diff
@@ -134,27 +117,89 @@ export const SidebarBottomBase = ({ api, notifications = [], status = {} }: Side
   const hasWarnings = warnings.length > 0;
   const hasErrors = errors.length > 0;
 
+  const updateTestProvider = useCallback(
+    (id: TestProviderId, update: Partial<TestProviderState>) =>
+      setTestProviders((state) => ({ ...state, [id]: { ...state[id], ...update } })),
+    []
+  );
+
   const onRunTests = useCallback(
-    (providerId?: string) => {
-      api.emit(TESTING_MODULE_RUN_ALL_REQUEST, { providerId });
+    (id: TestProviderId) => {
+      const startingState: Partial<TestProviderState> = {
+        cancelling: false,
+        running: true,
+        failed: false,
+      };
+      setTestProviders((state) => ({ ...state, [id]: { ...state[id], ...startingState } }));
+      api.experimental_updateStatus(id, (state = {}) =>
+        Object.fromEntries(Object.keys(state).map((key) => [key, null]))
+      );
+      api.emit(TESTING_MODULE_RUN_ALL_REQUEST, { providerId: id });
     },
     [api]
   );
+
+  const onCancelTests = useCallback(
+    (id: TestProviderId) => {
+      updateTestProvider(id, { cancelling: true });
+      api.emit(TESTING_MODULE_CANCEL_TEST_RUN_REQUEST, { providerId: id });
+    },
+    [api, updateTestProvider]
+  );
+
   const onSetWatchMode = useCallback(
     (providerId: string, watchMode: boolean) => {
+      updateTestProvider(providerId, { watching: watchMode });
       api.emit(TESTING_MODULE_WATCH_MODE_REQUEST, { providerId, watchMode });
     },
-    [api]
+    [api, updateTestProvider]
   );
 
+  useEffect(() => {
+    const wrapper = document.getElementById('sidebar-bottom');
```

Comment:
Can we use a const for the id and reuse it in the sidebar for the sidebar-bottom element as well?